pkg/argocd/repo.go

```diff
@@ -56,7 +52,30 @@ func CloneRepo(
 
 // -----------------------------------------------------------------------------
 
-func getRepoCreds(
+func GetKubeclientAndRepoCreds(
+	config *rest.Config,
+	argocdNs string,
+	repoUrl string,
+) (
+	kubeClient *kubernetes.Clientset,
+	creds *RepoCreds,
+	err error,
+) {
+	kubeClient, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get kubernetes client: %s", err)
+	}
+	creds, err = GetRepoCredsFromArgoCd(kubeClient, argocdNs, repoUrl)
+	if err != nil {
+		return nil, nil,
+			fmt.Errorf("failed to get repository credentials from argocd: %s", err)
+	}
+	return
+}
+
+// -----------------------------------------------------------------------------
+
+func GetRepoCredsFromArgoCd(
```

Comment:
Apologies for bringing this up right now, but instead of listing secrets, we can directly use the argocd library to fetch only the required repo like so: https://github.com/argoproj/argo-cd/blob/f93d678daebcf4c1424d6c309d969cf8150eb4de/cmd/argocd/commands/repo.go#L302
(Not a priority for now)